src/sentry/snuba/metrics/extraction.py

```diff
@@ -1226,14 +1245,11 @@ def _get_metric_type(function: str) -> str:
     @staticmethod
     def _parse_field(value: str) -> FieldParsingResult:
         try:
-            match = fields.is_function(value)
-            if not match:
-                raise InvalidSearchQuery(f"Invalid characters in field {value}")
-
-            function, _, arguments, alias = query_builder.parse_function(match)
-
+            function, arguments, alias = _parse_function(value)
             if function:
                 return FieldParsingResult(function=function, arguments=arguments, alias=alias)
+
+            # TODO: why is this here?
             column = query_builder.resolve_column(value)
```

Comment:
I am quite confused by this line of code, @armenzg could you help me to understand it?

Reply:
My apologies. I just saw this.

It's better to ask Kevan as he knows this well
https://github.com/getsentry/sentry/commit/47ccd85efd5f17c0602dce0723c39c71dd60a82f#diff-40193105c28bca8dc087b3b9cf5bc61343c35e6a36695435b2290a3f2946f536R1110